optimizely/client/factory.go

```diff
@@ -43,6 +42,68 @@ type OptimizelyFactory struct {
 	Datafile []byte
 }
 
+// OptionFunc is a type to a proper func
+type OptionFunc func(*OptimizelyClient, utils.ExecutionCtx)
+
+// GetClient gets client and sets some parameters
+func (f OptimizelyFactory) GetClient(clientOptions ...OptionFunc) *OptimizelyClient {
+	executionCtx := utils.NewCancelableExecutionCtx()
+	appClient := &OptimizelyClient{
+		isValid:      false,
+		executionCtx: executionCtx,
+	}
+	for _, opt := range clientOptions {
+		opt(appClient, executionCtx)
+	}
+	return appClient
```

Comment:
If we arrive at this step we should set isvalid to true for the client to be usable.

Reply:
As we discussed offline, let's get rid of the `isValid` property as it does not make sense to keep right now.

Reply:
done please check